plugin/storage/cassandra/spanstore/reader.go

```diff
@@ -299,19 +310,22 @@ func (s *SpanReader) findTraceIDs(ctx context.Context, traceQuery *spanstore.Tra
 		return traceIds, nil
 	}
 	if len(traceQuery.Tags) > 0 {
-		return s.queryByTagsAndLogs(ctx, traceQuery)
+		return s.queryByTagsAndLogs(ctx, traceQuery, s.tracer)
 	}
 	return s.queryByService(ctx, traceQuery)
 }
 
-func (s *SpanReader) queryByTagsAndLogs(ctx context.Context, tq *spanstore.TraceQueryParameters) (dbmodel.UniqueTraceIDs, error) {
-	span, ctx := startSpanForQuery(ctx, "queryByTagsAndLogs", queryByTag)
-	defer span.Finish()
+func (s *SpanReader) queryByTagsAndLogs(ctx context.Context, tq *spanstore.TraceQueryParameters, tracer trace.Tracer) (dbmodel.UniqueTraceIDs, error) {
```

Comment:
```suggestion
func (s *SpanReader) queryByTagsAndLogs(ctx context.Context, tq *spanstore.TraceQueryParameters) (dbmodel.UniqueTraceIDs, error) {
```
tracer is already available as `s.tracer`, no need to pass it